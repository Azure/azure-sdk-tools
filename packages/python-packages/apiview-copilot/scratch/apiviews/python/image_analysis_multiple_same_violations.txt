# Package is parsed using apiview-stub-generator(version:0.3.14), Python version: 3.12.9

namespace azure.ai.vision.imageanalysis

class azure.ai.vision.imageanalysis.ImageAnalysisClient(ImageAnalysisClient): implements ContextManager 
    def __init__(
        self, 
        endpoint: str, 
        credential: Union[AzureKeyCredential, TokenCredential], 
        *, 
        api_version: str = ..., 
        **kwargs: Any
    ) -> None

    @distributed_trace
    def analyze(
        self, 
        image_data: bytes, 
        visual_features: List[VisualFeatures], 
        gender_neutral_caption: Optional[bool] = ..., 
        language: Optional[str] = ..., 
        model_version: Optional[str] = ..., 
        smart_crops_aspect_ratios: Optional[List[float]] = ..., 
        **kwargs: Any
    ) -> ImageAnalysisResult

    @distributed_trace
    def analyze_from_url(
        self, 
        image_url: str, 
        visual_features: List[VisualFeatures], 
        gender_neutral_caption: Optional[bool] = ..., 
        language: Optional[str] = ..., 
        model_version: Optional[str] = ..., 
        smart_crops_aspect_ratios: Optional[List[float]] = ..., 
        **kwargs: Any
    ) -> ImageAnalysisResult

    def close(self) -> None

    def send_request(
        self, 
        request: HttpRequest, 
        *, 
        stream: bool = False, 
        **kwargs: Any
    ) -> HttpResponse

namespace azure.ai.vision.imageanalysis.aio

class azure.ai.vision.imageanalysis.aio.ImageAnalysisClient(ImageAnalysisClient): implements AsyncContextManager 
    def __init__(
        self, 
        endpoint: str, 
        credential: Union[AzureKeyCredential, AsyncTokenCredential], 
        *, 
        api_version: str = ..., 
        **kwargs: Any
    ) -> None

    @distributed_trace_async
    async def analyze(
        self, 
        image_data: bytes, 
        visual_features: List[VisualFeatures], 
        gender_neutral_caption: Optional[bool] = ..., 
        language: Optional[str] = ..., 
        model_version: Optional[str] = ..., 
        smart_crops_aspect_ratios: Optional[List[float]] = ..., 
        **kwargs: Any
    ) -> ImageAnalysisResult

    @distributed_trace_async
    async def analyze_from_url(
        self, 
        image_url: str, 
        visual_features: List[VisualFeatures], 
        gender_neutral_caption: Optional[bool] = ..., 
        language: Optional[str] = ..., 
        model_version: Optional[str] = ..., 
        smart_crops_aspect_ratios: Optional[List[float]] = ..., 
        **kwargs: Any
    ) -> ImageAnalysisResult

    async def close(self) -> None

    def send_request(
        self, 
        request: HttpRequest, 
        *, 
        stream: bool = False, 
        **kwargs: Any
    ) -> Awaitable[AsyncHttpResponse]

namespace azure.ai.vision.imageanalysis.models

class azure.ai.vision.imageanalysis.models.CaptionResult(MutableMapping[str, Any]):
    ivar confidence: float
    ivar text: str

    @overload
    def __init__(
        self, 
        *, 
        confidence: float, 
        text: str
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None

class azure.ai.vision.imageanalysis.models.CropRegion(MutableMapping[str, Any]):
    ivar aspect_ratio: float
    ivar bounding_box: ImageBoundingBox

    @overload
    def __init__(
        self, 
        *, 
        aspect_ratio: float, 
        bounding_box: ImageBoundingBox
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None

namespace azure.ai.vision.imageanalysis.models

class azure.ai.vision.imageanalysis.models.DenseCaption(MutableMapping[str, Any]):
    ivar bounding_box: ImageBoundingBox
    ivar confidence: float
    ivar text: str

    @overload
    def __init__(
        self, 
        *, 
        bounding_box: ImageBoundingBox, 
        confidence: float, 
        text: str
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None


class azure.ai.vision.imageanalysis.models.DenseCaptionsResult(MutableMapping[str, Any]):
    ivar list: List[DenseCaption]

    @overload
    def __init__(
        self, 
        *, 
        list: List[DenseCaption]
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None


class azure.ai.vision.imageanalysis.models.DetectedTextBlock(MutableMapping[str, Any]):
    ivar lines: List[DetectedTextLine]

    @overload
    def __init__(
        self, 
        *, 
        lines: List[DetectedTextLine]
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None

class azure.ai.vision.imageanalysis.models.DetectedTextLine(MutableMapping[str, Any]):
    ivar bounding_polygon: List[ImagePoint]
    ivar text: str
    ivar words: List[DetectedTextWord]

    @overload
    def __init__(
        self, 
        *, 
        bounding_polygon: List[ImagePoint], 
        text: str, 
        words: List[DetectedTextWord]
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None


class azure.ai.vision.imageanalysis.models.DetectedTextWord(MutableMapping[str, Any]):
    ivar bounding_polygon: List[ImagePoint]
    ivar confidence: float
    ivar text: str

    @overload
    def __init__(
        self, 
        *, 
        bounding_polygon: List[ImagePoint], 
        confidence: float, 
        text: str
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None


class azure.ai.vision.imageanalysis.models.ImageAnalysisResult(MutableMapping[str, Any]):
    ivar caption: Optional[CaptionResult]
    ivar dense_captions: Optional[DenseCaptionsResult]
    ivar metadata: ImageMetadata
    ivar model_version: str
    ivar objects: Optional[ObjectsResult]
    ivar people: Optional[PeopleResult]
    ivar read: Optional[ReadResult]
    ivar smart_crops: Optional[SmartCropsResult]
    ivar tags: Optional[TagsResult]

    @overload
    def __init__(
        self, 
        *, 
        caption: Optional[CaptionResult] = ..., 
        dense_captions: Optional[DenseCaptionsResult] = ..., 
        metadata: ImageMetadata, 
        model_version: str, 
        objects: Optional[ObjectsResult] = ..., 
        people: Optional[PeopleResult] = ..., 
        read: Optional[ReadResult] = ..., 
        smart_crops: Optional[SmartCropsResult] = ..., 
        tags: Optional[TagsResult] = ...
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None


class azure.ai.vision.imageanalysis.models.ImageBoundingBox(MutableMapping[str, Any]):
    ivar height: int
    ivar width: int
    ivar x: int
    ivar y: int

    @overload
    def __init__(
        self, 
        *, 
        height: int, 
        width: int, 
        x: int, 
        y: int
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None

class azure.ai.vision.imageanalysis.models.ImageMetadata(MutableMapping[str, Any]):
    ivar height: int
    ivar width: int

    @overload
    def __init__(
        self, 
        *, 
        height: int, 
        width: int
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None


class azure.ai.vision.imageanalysis.models.ImagePoint(MutableMapping[str, Any]):
    ivar x: int
    ivar y: int

    @overload
    def __init__(
        self, 
        *, 
        x: int, 
        y: int
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None


class azure.ai.vision.imageanalysis.models.ObjectsResult(MutableMapping[str, Any]):
    ivar list: List[DetectedObject]

    @overload
    def __init__(
        self, 
        *, 
        list: List[DetectedObject]
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None


class azure.ai.vision.imageanalysis.models.PeopleResult(MutableMapping[str, Any]):
    ivar list: List[DetectedPerson]

    @overload
    def __init__(
        self, 
        *, 
        list: List[DetectedPerson]
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None


class azure.ai.vision.imageanalysis.models.ReadResult(MutableMapping[str, Any]):
    ivar blocks: List[DetectedTextBlock]

    @overload
    def __init__(
        self, 
        *, 
        blocks: List[DetectedTextBlock]
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None


class azure.ai.vision.imageanalysis.models.SmartCropsResult(MutableMapping[str, Any]):
    ivar list: List[CropRegion]

    @overload
    def __init__(
        self, 
        *, 
        list: List[CropRegion]
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None

class azure.ai.vision.imageanalysis.models.TagsResult(MutableMapping[str, Any]):
    ivar list: List[DetectedTag]

    @overload
    def __init__(
        self, 
        *, 
        list: List[DetectedTag]
    )

    @overload
    def __init__(self, mapping: Mapping[str, Any])

    def __init__(
        self, 
        *args: Any, 
        **kwargs: Any
    ) -> None

class azure.ai.vision.imageanalysis.models.VisualFeatures(str, Enum):
    CAPTION = "caption"
    DENSE_CAPTIONS = "denseCaptions"
    OBJECTS = "objects"
    PEOPLE = "people"
    READ = "read"
    SMART_CROPS = "smartCrops"
    TAGS = "tags"