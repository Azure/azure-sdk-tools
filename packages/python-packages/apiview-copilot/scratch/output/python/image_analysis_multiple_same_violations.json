{
    "comments": [
        {
            "rule_ids": [],
            "line_no": 1,
            "bad_code": "",
            "suggestion": null,
            "comment": "Here is a summary of the service described by this APIView:\n\n**Purpose**  \nThis API provides a service for analyzing images by extracting structured insights such as captions, dense captions, detected objects and people, text recognition, smart cropping suggestions, and tags. Its functionality is centered on processing image content and returning rich analytical metadata.\n\n**API Version**  \nThe client classes accept an `api_version` parameter to specify the version of the service to use, with the latest version expected to be managed by the service endpoint.\n\n**Client Classes**  \nThe API exposes two primary client classes: the synchronous `azure.ai.vision.imageanalysis.ImageAnalysisClient` and the asynchronous `azure.ai.vision.imageanalysis.aio.ImageAnalysisClient`. Both clients offer equivalent functionality for submitting image analysis requests.\n\n**Functional Overview**  \nThe clients enable users to analyze images by either providing raw image bytes or specifying an image URL. They support a range of visual features invoked through the analysis operations, which perform actions akin to the read side of CRUD by interpreting and returning detailed image analysis results. Parameters exist to adjust options such as gender neutrality in captions, language, specific model versions, and custom aspect ratios for smart crops, while underlying methods handle HTTP request interactions with the service.",
            "source": "summary"
        },
        {
            "rule_ids": [
                "python_design.html#python-client-optional-arguments-keyword-only"
            ],
            "line_no": 16,
            "bad_code": "def analyze(",
            "suggestion": "def analyze(self, image_data: bytes, visual_features: List[VisualFeatures], *, gender_neutral_caption: Optional[bool] = ..., language: Optional[str] = ..., model_version: Optional[str] = ..., smart_crops_aspect_ratios: Optional[List[float]] = ..., **kwargs: Any) -> ImageAnalysisResult",
            "comment": "The synchronous analyze method’s optional parameters should be keyword-only. Insert a '*' after the required parameters to enforce this.",
            "source": "guideline"
        },
        {
            "rule_ids": [
                "python_design.html#python-client-optional-arguments-keyword-only"
            ],
            "line_no": 28,
            "bad_code": "def analyze_from_url(",
            "suggestion": "def analyze_from_url(self, image_url: str, visual_features: List[VisualFeatures], *, gender_neutral_caption: Optional[bool] = ..., language: Optional[str] = ..., model_version: Optional[str] = ..., smart_crops_aspect_ratios: Optional[List[float]] = ..., **kwargs: Any) -> ImageAnalysisResult",
            "comment": "The synchronous analyze_from_url method’s optional parameters should be keyword-only. Add a '*' to separate required parameters from the optional ones.",
            "source": "guideline"
        },
        {
            "rule_ids": [
                "python_design.html#python-client-optional-arguments-keyword-only"
            ],
            "line_no": 62,
            "bad_code": "async def analyze(",
            "suggestion": "async def analyze(self, image_data: bytes, visual_features: List[VisualFeatures], *, gender_neutral_caption: Optional[bool] = ..., language: Optional[str] = ..., model_version: Optional[str] = ..., smart_crops_aspect_ratios: Optional[List[float]] = ..., **kwargs: Any) -> ImageAnalysisResult",
            "comment": "For the async analyze method, optional parameters must be keyword-only. Prepend a '*' after the required parameters.",
            "source": "guideline"
        },
        {
            "rule_ids": [
                "python_design.html#python-client-optional-arguments-keyword-only"
            ],
            "line_no": 74,
            "bad_code": "async def analyze_from_url(",
            "suggestion": "async def analyze_from_url(self, image_url: str, visual_features: List[VisualFeatures], *, gender_neutral_caption: Optional[bool] = ..., language: Optional[str] = ..., model_version: Optional[str] = ..., smart_crops_aspect_ratios: Optional[List[float]] = ..., **kwargs: Any) -> ImageAnalysisResult",
            "comment": "The async analyze_from_url method’s optional parameters should be keyword-only. Adding '*' after required arguments will enforce this.",
            "source": "guideline"
        },
        {
            "rule_ids": [],
            "line_no": 166,
            "bad_code": "ivar list: List[DenseCaption]",
            "suggestion": "ivar captions: List[DenseCaption]",
            "comment": "Avoid using the built-in name 'list' for an attribute; use a more descriptive name like 'captions'.",
            "source": "generic"
        },
        {
            "rule_ids": [],
            "line_no": 172,
            "bad_code": "        list: List[DenseCaption]",
            "suggestion": "        captions: List[DenseCaption]",
            "comment": "Rename parameter 'list' to 'captions' in the overload to prevent shadowing the built-in type.",
            "source": "generic"
        },
        {
            "rule_ids": [],
            "line_no": 358,
            "bad_code": "ivar list: List[DetectedObject]",
            "suggestion": "ivar objects: List[DetectedObject]",
            "comment": "Rename attribute 'list' to 'objects' to clarify its content and avoid shadowing the built-in.",
            "source": "generic"
        },
        {
            "rule_ids": [],
            "line_no": 364,
            "bad_code": "        list: List[DetectedObject]",
            "suggestion": "        objects: List[DetectedObject]",
            "comment": "Change parameter name from 'list' to 'objects' to avoid conflict with the built-in type.",
            "source": "generic"
        },
        {
            "rule_ids": [],
            "line_no": 378,
            "bad_code": "ivar list: List[DetectedPerson]",
            "suggestion": "ivar people: List[DetectedPerson]",
            "comment": "Rename attribute 'list' to 'people' to better reflect its semantics and avoid using a built-in name.",
            "source": "generic"
        },
        {
            "rule_ids": [],
            "line_no": 384,
            "bad_code": "        list: List[DetectedPerson]",
            "suggestion": "        people: List[DetectedPerson]",
            "comment": "Rename the parameter from 'list' to 'people' to prevent shadowing the built-in and improve clarity.",
            "source": "generic"
        },
        {
            "rule_ids": [],
            "line_no": 418,
            "bad_code": "ivar list: List[CropRegion]",
            "suggestion": "ivar smart_crops: List[CropRegion]",
            "comment": "Use a more descriptive attribute name than 'list' (e.g. 'smart_crops') to avoid overriding the built-in.",
            "source": "generic"
        },
        {
            "rule_ids": [],
            "line_no": 424,
            "bad_code": "        list: List[CropRegion]",
            "suggestion": "        smart_crops: List[CropRegion]",
            "comment": "Rename parameter 'list' to 'smart_crops' in the overload to avoid conflict with the built-in name.",
            "source": "generic"
        },
        {
            "rule_ids": [],
            "line_no": 437,
            "bad_code": "ivar list: List[DetectedTag]",
            "suggestion": "ivar tags: List[DetectedTag]",
            "comment": "Replace the generic 'list' with 'tags' to better describe the content and avoid shadowing the built-in.",
            "source": "generic"
        },
        {
            "rule_ids": [],
            "line_no": 443,
            "bad_code": "        list: List[DetectedTag]",
            "suggestion": "        tags: List[DetectedTag]",
            "comment": "Change the parameter name from 'list' to 'tags' to prevent using the built-in name and improve readability.",
            "source": "generic"
        }
    ]
}