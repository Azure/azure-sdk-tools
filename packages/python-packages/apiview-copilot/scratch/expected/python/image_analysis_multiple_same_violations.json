{
  "status": "Error",
  "violations": [
    {
      "rule_ids": [
        "python_design.html#python-client-optional-arguments-keyword-only"
      ],
      "line_no": 16,
      "bad_code": "def analyze(",
      "suggestion": "def analyze(self, image_data: bytes, visual_features: List[VisualFeatures], *, gender_neutral_caption: Optional[bool] = ..., language: Optional[str] = ..., model_version: Optional[str] = ..., smart_crops_aspect_ratios: Optional[List[float]] = ..., **kwargs: Any) -> ImageAnalysisResult",
      "comment": "Optional operation-specific parameters should be provided as keyword-only arguments. Inserting the '*' after the required parameters forces callers to use parameter names."
    },
    {
      "rule_ids": [
        "python_design.html#python-client-optional-arguments-keyword-only"
      ],
      "line_no": 28,
      "bad_code": "def analyze_from_url(",
      "suggestion": "def analyze_from_url(self, image_url: str, visual_features: List[VisualFeatures], *, gender_neutral_caption: Optional[bool] = ..., language: Optional[str] = ..., model_version: Optional[str] = ..., smart_crops_aspect_ratios: Optional[List[float]] = ..., **kwargs: Any) -> ImageAnalysisResult",
      "comment": "Optional parameters should be keyword-only; add '*' after required parameters in the method signature."
    },
    {
      "rule_ids": [
        "python_design.html#python-client-optional-arguments-keyword-only"
      ],
      "line_no": 62,
      "bad_code": "async def analyze(",
      "suggestion": "async def analyze(self, image_data: bytes, visual_features: List[VisualFeatures], *, gender_neutral_caption: Optional[bool] = ..., language: Optional[str] = ..., model_version: Optional[str] = ..., smart_crops_aspect_ratios: Optional[List[float]] = ..., **kwargs: Any) -> ImageAnalysisResult",
      "comment": "For async methods, optional parameters must also be keyword-only. Insert '*' to force keyword-only usage."
    },
    {
      "rule_ids": [
        "python_design.html#python-client-optional-arguments-keyword-only"
      ],
      "line_no": 74,
      "bad_code": "async def analyze_from_url(",
      "suggestion": "async def analyze_from_url(self, image_url: str, visual_features: List[VisualFeatures], *, gender_neutral_caption: Optional[bool] = ..., language: Optional[str] = ..., model_version: Optional[str] = ..., smart_crops_aspect_ratios: Optional[List[float]] = ..., **kwargs: Any) -> ImageAnalysisResult",
      "comment": "Optional parameters should be made keyword-only by inserting '*' after the required parameters."
    }
  ]
}
